src/test/java/org/opentripplanner/middleware/ApiUserFlowTest.java

```diff
@@ -0,0 +1,172 @@
+ * configured for your operating environment e.g. C:\Users\<username>\.aws\credentials in Windows or Mac OS equivalent.
+ *
+ * The following environment variable must be set for these tests to run: - RUN_E2E=true
+ *
+ * Auth0 must be correctly configured as described here: https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow
+ */
+public class ApiUserFlowTest {
+    private static ApiUser apiUser;
+    private static OtpUser otpUser;
+
+    /**
+     * Create an {@link ApiUser} and an {@link AdminUser} prior to unit tests
+     */
+    @BeforeAll
+    public static void setUp() throws IOException {
+        assumeTrue(getBooleanEnvVar("RUN_E2E"));
+        OtpMiddlewareTest.setUp();
+        TestUtils.mockOtpServer();
+        // As a pre-condition, create an API User with API key.
+        apiUser = PersistenceUtil.createApiUser(String.format("test-%s@example.com", UUID.randomUUID().toString()));
+        apiUser.createApiKey(DEFAULT_USAGE_PLAN_ID, true);
+
+        otpUser = new OtpUser();
+        otpUser.email = String.format("test-%s@example.com", UUID.randomUUID().toString());
+        otpUser.hasConsentedToTerms = true;
+        otpUser.storeTripHistory = true;
+
+        if (!authDisabled()) {
+            // create Auth0 users for apiUser and optUser.
+            try {
+                User auth0User = createAuth0UserForEmail(apiUser.email, TestUtils.TEMP_AUTH0_USER_PASSWORD);
+                // update api user with valid auth0 user ID (so the Auth0 delete works)
+                apiUser.auth0UserId = auth0User.getId();
+                Persistence.apiUsers.replace(apiUser.id, apiUser);
+
+                auth0User = createAuth0UserForEmail(otpUser.email, TestUtils.TEMP_AUTH0_USER_PASSWORD);
+                // update otp user with valid auth0 user ID (so the Auth0 delete works)
+                otpUser.auth0UserId = auth0User.getId();
+            } catch (Auth0Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * Delete the users if they were not already deleted during the test script.
+     */
+    @AfterAll
+    public static void tearDown() {
+        assumeTrue(getBooleanEnvVar("RUN_E2E"));
+        apiUser = Persistence.apiUsers.getById(apiUser.id);
+        if (apiUser != null) apiUser.delete();
+        otpUser = Persistence.otpUsers.getById(otpUser.id);
+        if (otpUser != null) otpUser.delete();
+    }
+
+    /**
+     * Tests to confirm that an otp user, related monitored trip and plan can be created and deleted leaving no orphaned
+     * records. This also includes Auth0 users if auth is enabled.
+     */
+    @Test
+    public void canSimulateApiUserFlow() {
+        assumeTrue(getBooleanEnvVar("RUN_E2E"));
+
+        // create otp user as api user
+        HttpResponse<String> createUserResponse = mockAuthenticatedPost("api/secure/user",
```

Comment:
This is tricky. In order to complete, this test needs to obtain a valid Auth0 `accessToken`.
The fake `TokenHolder` produced by `Auth0UsersTest` does not provide one, and the call to the Auth0 management API fails as a result.

Reply:
Fixed